### ISSUE TITLE
Gray screen

### ISSUE THREAD

**Issue by Gatoz97** [iwrqk/iwrqk#7]
On the most recent version non beta 1.1.0.
The app consistenly gray screens when backing from a post or switching tabs, its basically unusable.
This already happened in the beta version but this time is unavoidable.
I am using the v8a version if that matters.

**Comment by Kelniimir** [comment]
> On the most recent version non beta 1.1.0.
> The app consistenly gray screens when backing from a post or switching tabs, its basically unusable.
> This already happened in the beta version but this time is unavoidable.
> I am using the v8a version if that matters.

I confirm same behavior on Android 13, Samsung OneUI 5.1, on arm64-v8a as well.

**Comment by iwrneko** [comment]
Based on your description, I cannot reproduce this problem. Actually, it works fine on my device (Pixel AVD Android 13, LG Android 10 and even WSA).
You can check if there are logs in the log directory `/data/data/com.iwrqk.app/app_flutter/logs/` (this is experimental, and some devices still cannot generate logs) and then upload it. If not, notify me and I will provide a debug version for you.

**Comment by iwrneko** [comment]
[app-arm64-v8a-debug.apk](https://mega.nz/file/Uz9lmCqL#h1ynSsrKazUc5fJu00q4DY3514D5bHHpCLt7NkEHqxM)

**Comment by iwrneko** [comment]
> > On the most recent version non beta 1.1.0.
> > The app consistenly gray screens when backing from a post or switching tabs, its basically unusable.
> > This already happened in the beta version but this time is unavoidable.
> > I am using the v8a version if that matters.
> 
> I confirm same behavior on Android 13, Samsung OneUI 5.1, on arm64-v8a as well.

Based on your description, I cannot reproduce this problem. Actually, it works fine on my device (Pixel AVD Android 13, LG Android 10 and even WSA).
You can check if there are logs in the log directory `/data/data/com.iwrqk.app/app_flutter/logs/` (this is experimental, and some devices still cannot generate logs) and then upload it. If not, I will provide a debug version for you.
In debug mode, the gray screen will be replaced by a red screen where print the error information. This is need for us to find out what happen.
[app-arm64-v8a-debug.apk](https://mega.nz/file/Uz9lmCqL#h1ynSsrKazUc5fJu00q4DY3514D5bHHpCLt7NkEHqxM)

**Comment by Kelniimir** [comment]
> > > On the most recent version non beta 1.1.0.
> > > The app consistenly gray screens when backing from a post or switching tabs, its basically unusable.
> > > This already happened in the beta version but this time is unavoidable.
> > > I am using the v8a version if that matters.
> > 
> > I confirm same behavior on Android 13, Samsung OneUI 5.1, on arm64-v8a as well.
> 
> Based on your description, I cannot reproduce this problem. Actually, it works fine on my device (Pixel AVD Android 13, LG Android 10 and even WSA).
> You can check if there are logs in the log directory `/data/data/com.iwrqk.app/app_flutter/logs/` (this is experimental, and some devices still cannot generate logs) and then upload it. If not, I will provide a debug version for you.
> In debug mode, the gray screen will be replaced by a red screen where print the error information. This is need for us to find out what happen.
> [app-arm64-v8a-debug.apk](https://mega.nz/file/Uz9lmCqL#h1ynSsrKazUc5fJu00q4DY3514D5bHHpCLt7NkEHqxM)

Checked.
On Debug version, issue isn't reproducing, after 20 minutes of attempts.
Deleted and installed 1.1.0 again.
Application refused to go pass loading screen (with blue waves on bottom, prior to "Getting started" configuration and login screen).
Deleted and reinstalled 1.1.0.
After that issue isn't reproducing again.

Even though I couldn't "upgrade" and was always reinstalling, I noticed sometimes Samsung keep data after deletion, and sometimes they don't. There is slight possibility some files left from Alpha / Beta influenced release version?

TL;DR: Issue isn't reproducing any longer, andter deletion and reinstall of few versions.

Maybe original topic author would have more luck with tracking it.

**Comment by iwrneko** [comment]
> > > > On the most recent version non beta 1.1.0.
> > > > The app consistenly gray screens when backing from a post or switching tabs, its basically unusable.
> > > > This already happened in the beta version but this time is unavoidable.
> > > > I am using the v8a version if that matters.
> > > 
> > > 
> > > I confirm same behavior on Android 13, Samsung OneUI 5.1, on arm64-v8a as well.
> > 
> > 
> > Based on your description, I cannot reproduce this problem. Actually, it works fine on my device (Pixel AVD Android 13, LG Android 10 and even WSA).
> > You can check if there are logs in the log directory `/data/data/com.iwrqk.app/app_flutter/logs/` (this is experimental, and some devices still cannot generate logs) and then upload it. If not, I will provide a debug version for you.
> > In debug mode, the gray screen will be replaced by a red screen where print the error information. This is need for us to find out what happen.
> > [app-arm64-v8a-debug.apk](https://mega.nz/file/Uz9lmCqL#h1ynSsrKazUc5fJu00q4DY3514D5bHHpCLt7NkEHqxM)
> 
> Checked. On Debug version, issue isn't reproducing, after 20 minutes of attempts. Deleted and installed 1.1.0 again. Application refused to go pass loading screen (with blue waves on bottom, prior to "Getting started" configuration and login screen). Deleted and reinstalled 1.1.0. After that issue isn't reproducing again.
> 
> Even though I couldn't "upgrade" and was always reinstalling, I noticed sometimes Samsung keep data after deletion, and sometimes they don't. There is slight possibility some files left from Alpha / Beta influenced release version?
> 
> TL;DR: Issue isn't reproducing any longer, after deletion and reinstall of few versions (release, and Debug).
> 
> Maybe original topic author would have more luck with tracking it.

I think it's strange. After all, developing can also be considered as an update on my device, but it works fine on my device. I guess it is might caused by the new dependency 'package_info_plus' added by the check for update function introduced in the new version. Actually, this is a very common and low-level dependency, and many dependencies rely on it, and other dependencies may rely on updated or older versions of the 'package_info_plus'. This may cause the cache generated by the old version to make it happening in the new version.
`Application refused to go pass loading screen` also proves this point that the loading screen is taking the task of checking for update. But without many details, I cannot figure out how exactly this happening, the above are all my speculations.
